Dockerfile

```diff
@@ -23,11 +21,40 @@ RUN chmod +x ./entrypoint-prod.sh
 ENTRYPOINT ["sh", "entrypoint-prod.sh"]
 
 FROM base-image as todo-dev
-# copy app code, test code and run flask
-# COPY entrypoint-dev.sh ./*.py ./
-# COPY ./templates/ ./templates/
-# COPY ./tests/ ./tests/
+# copy entrypoint script and run
 COPY entrypoint-dev.sh ./
 EXPOSE 5000
 RUN chmod +x ./entrypoint-dev.sh
 ENTRYPOINT ["sh", "entrypoint-dev.sh"]
+
+FROM base-image as todo-test
+# Install chrome
+RUN apt-get update && \
+    apt-get install -y gnupg wget curl unzip --no-install-recommends && \
+    wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - && \
+    echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list && \
+    apt-get update -y && \
+    apt-get install -y google-chrome-stable
+# Install Chromium WebDriver
+RUN LATEST=`curl -sSL https://chromedriver.storage.googleapis.com/LATEST_RELEASE` &&\
+    echo "Installing chromium webdriver version ${LATEST}" &&\
+    curl -sSL https://chromedriver.storage.googleapis.com/${LATEST}/chromedriver_linux64.zip -o chromedriver_linux64.zip &&\
+    apt-get install unzip -y &&\
+    unzip ./chromedriver_linux64.zip
+# Install Firefox and Geckodriver (needs work, commenting out)
+# RUN GECKODRIVER_VERSION=`curl https://github.com/mozilla/geckodriver/releases/latest | grep -Po 'v[0-9]+.[0-9]+.[0-9]+'` && \
+#     wget https://github.com/mozilla/geckodriver/releases/download/$GECKODRIVER_VERSION/geckodriver-$GECKODRIVER_VERSION-linux64.tar.gz && \
+#     tar -zxf geckodriver-$GECKODRIVER_VERSION-linux64.tar.gz -C ./ && \
+#     chmod +x ./geckodriver && \
+#     rm geckodriver-$GECKODRIVER_VERSION-linux64.tar.gz
+# RUN FIREFOX_SETUP=firefox-setup.tar.bz2 && \
+#     apt-get purge firefox && \
+#     wget -O $FIREFOX_SETUP "https://download.mozilla.org/?product=firefox-latest&os=linux64" && \
+#     tar xjf $FIREFOX_SETUP -C /opt/ && \
+#     ln -s /opt/firefox/firefox ./ && \
+#     rm $FIREFOX_SETUP
```

Comment:
Delete commented out code (also the last line in the entrypoint sh file).

Is there a reason you have both Firefox and Chrome versions of the e2e test?

Reply:
I have deleted the commented out code.
I have 2 versions as I have not yet got around to working out how to merge them into 1 and I wanted to try both browsers.  I can't get firefox to run headless though, it always launches the browser, so removed from Travis build for now... will come back to it at some point